src/Umbraco.Web/Search/ExamineComposer.cs

```diff
@@ -32,6 +37,12 @@ public override void Compose(Composition composition)
             composition.Register<IndexRebuilder>(Lifetime.Singleton);
             composition.RegisterUnique<IUmbracoIndexConfig, UmbracoIndexConfig>();
             composition.RegisterUnique<IUmbracoIndexesCreator, UmbracoIndexesCreator>();
+            composition.RegisterUnique<ILoggingService, UmbracoExamineLoggingService>();
+            foreach (Type factory in composition.TypeLoader.GetTypes<IDirectoryFactory>())
```

Comment:
We should not type scan here. We only ever type scan for a single type: IDiscoverable else there is a lot more overhead. Each directory factory should just be explicitly registered.